test/test-ui-action-button.js

```diff
@@ -716,7 +717,11 @@ exports['test button tab state'] = function*(assert) {
     'tab badgeColor inherited from global');
 
   // check the node properties
-  yield wait();
+  yield new Promise(resolve => {
+    let target = {};
+    once(target, "ready", resolve);
+    emit(target, "ready");
+  });
```

Comment:
Well I'm not sure tht it is better, but it seems like using setImmediate is failing us.